### ISSUE TITLE
installer: wrong message when using UV_INSTALL_DIR

### ISSUE THREAD

**Issue by ilyagr** [astral-sh/uv#8300]
I ran

```console
$ curl -LsSf https://astral.sh/uv/install.sh | env INSTALLER_NO_MODIFY_PATH=1 UV_INSTALL_DIR=~/.local/bin sh
downloading uv 0.4.23 aarch64-apple-darwin
installing to ~/.local/bin/bin
  uv
  uvx
everything's installed!
```

Note the message saying "installing to ~/.local/bin/bin". Actually, the install seemed to correctly go to `~/.local/bin`, it's just the message that's wrong.

**Comment by ilyagr** [comment]
On my system (Mac + fish), there is a helpful `printenv` command that shows:

```console
$ a=~ printenv a
/Users/ilyagr
$ env a=~ printenv a
~
```

This makes no sense to me. I filed a question at https://github.com/fish-shell/fish-shell/issues/10794 about that.

**Comment by zanieb** [comment]
> Actually, the install ended up going in /Users/ilyagr/tmp/~/.local/bin/bin/uv (with a literal ~ in the name), which has several problems with it. I was in /Users/ilyagr/tmp/ when running the command.

Yeah this seems like a shell issue, that's weird. I'm confused too honestly.

Sounds like you want `UV_UNMANAGED_INSTALL` if you don't want to create the `bin` directory? it's setup this way because we might need to install other things than binaries.

**Comment by ilyagr** [comment]
Bash acts the way I'd expect, so I thought it was a fish issue. However, `zsh` and `dash` both act like `fish`, so I guess the `env` trick I suggested a while ago is just not great. Sorry.

**Comment by zanieb** [comment]
The documentation only recommends an absolute path though, right?
